### PR TITLE
Show techmd

### LINKED ISSUE #1824: Show techmd for an item
Show techmd for an item from the techmd-service.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #1824

## Why was this change made?
To show technical metadata from the technical-metadata-service.

## Was the documentation updated?
No.

### CODE REVIEW

**Comment by justinlittman** [comment]
image

**Comment by justinlittman** [comment]
Neither the code or the UI is pretty, but I'd suggest that good enough for now:
image

File: app/controllers/catalog_controller.rb [hunk 1]
```diff
@@ -276,4 +279,15 @@ def limit_facets_on_home_page
...
+
+  def techmd_for(druid)
+    resp = Faraday.get("#{Settings.tech_md_service.url}/v1/technical-metadata/druid/#{druid}") do |req|
+      req.headers['Accept'] = 'application/json'
+      req.headers['Authorization'] = "Bearer #{Settings.tech_md_service.token}"
+    end
+    return JSON.parse(resp.body) if resp.status == 200
+    return [] if resp.status == 404
+
+    raise "Unexpected response from technical-metadata-service for #{druid}: #{resp.status}"
```
Comment (mjgiarlo) [inline app/controllers/catalog_controller.rb]: Also worth including the `resp.body` in the exception?

File: app/controllers/catalog_controller.rb [hunk 2]
```diff
@@ -276,4 +279,15 @@ def limit_facets_on_home_page
       v.include_in_request = false if v.home == false
     end
   end
+
+  def techmd_for(druid)
```
Comment (jcoyne) [inline app/controllers/catalog_controller.rb]: Please extract this method from the controller to a service class.

File: app/controllers/catalog_controller.rb [hunk 3]
```diff
@@ -202,6 +202,9 @@ def show
       solr_document: @document,
       object: @obj
     )
+
+    @techmd = techmd_for(params[:id])
```
Comment (jcoyne) [inline app/controllers/catalog_controller.rb]:
```suggestion
    @techmd = techmd_for(params[:id]) if params[:beta]
```

File: app/views/catalog/_techmd_default.html.erb [hunk 4]
```diff
@@ -0,0 +1,64 @@
+<% if params[:beta] %>
```
Comment (jcoyne) [inline app/views/catalog/_techmd_default.html.erb]: Remove this line (conditional was added in the controller)

**Pull request merged**